Analysis/Tasks/PWGGA/gammaConversionsMC.cxx

```diff
@@ -0,0 +1,324 @@
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+
+#include "AnalysisDataModel/StrangenessTables.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h" // for BigTracks
+
+#include "AnalysisDataModel/PID/PIDResponse.h"
+#include "AnalysisDataModel/PID/PIDTPC.h"
+
+#include <TH1F.h>
+#include <TVector3.h>
+
+#include <cmath>
+#include <iostream>
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+using tracksAndTPCInfo = soa::Join<aod::Tracks, aod::TracksExtra, aod::pidTPCEl, aod::pidTPCPi, aod::McTrackLabels>;
+
+struct GammaConversionsMc {
+
+  Configurable<float> fSinglePtCut{"fSinglePtCut", 0.04, "minimum daughter track pt"};
+
+  Configurable<float> fEtaCut{"fEtaCut", 0.8, "accepted eta range"};
+
+  Configurable<float> fMinR{"fMinR", 5., "minimum conversion radius of the V0s"};
+  Configurable<float> fMaxR{"fMaxR", 180., "maximum conversion radius of the V0s"};
+
+  Configurable<float> fPIDnSigmaBelowElectronLine{"fPIDnSigmaBelowElectronLine", -3., "minimum sigma electron PID for V0 daughter tracks"};
+
+  Configurable<float> fPIDnSigmaAboveElectronLine{"fPIDnSigmaAboveElectronLine", 3., "maximum sigma electron PID for V0 daughter tracks"};
+
+  Configurable<float> fPIDnSigmaAbovePionLine{"fPIDnSigmaAbovePionLine", 3., "minimum sigma to be over the pion line for low momentum tracks"}; //case 4: 3.0sigma, 1.0 sigma at high momentum
+
+  Configurable<float> fPIDnSigmaAbovePionLineHighP{"fPIDnSigmaAbovePionLineHighP", 1., "minimum sigma to be over the pion line for high momentum tracks"};
+
+  Configurable<float> fPIDMinPnSigmaAbovePionLine{"fPIDMinPnSigmaAbovePionLine", 0.4, "minimum track momentum to apply any pion rejection"}; //case 7:  // 0.4 GeV
+
+  Configurable<float> fPIDMaxPnSigmaAbovePionLine{"fPIDMaxPnSigmaAbovePionLine", 8., "border between low and high momentum pion rejection"}; //case 7:  // 8. GeV
+
+  Configurable<float> fMinTPCNClsOverFindable{"fMinTPCNClsFoundOverFindable", 0.6, "minimum ratio found tpc clusters over findable"}; //case 9:  // 0.6
+
+  Configurable<float> fQtPtMax{"fQtPtMax", 0.11, "up to fQtMax, multiply the pt of the V0s by this value to get the maximum qt "};
+
+  Configurable<float> fQtMax{"fQtMax", 0.040, "maximum qt"};
+  Configurable<float> fMaxPhotonAsymmetry{"fMaxPhotonAsymmetry", 0.95, "maximum photon asymetry"};
+  Configurable<float> fPsiPairCut{"fPsiPairCut", 0.1, "maximum psi angle of the track pair"};
+
+  Configurable<float> fCosPAngleCut{"fCosPAngleCut", 0.85, "mimimum cosinus of the pointing angle"}; // case 4
+
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"IsPhotonSelected", "IsPhotonSelected", {HistType::kTH1F, {{12, -0.5f, 11.5f}}}},
+
+      {"beforeCuts/hPtRec_before", "hPtRec_before", {HistType::kTH1F, {{100, 0.0f, 25.0f}}}},
```

Comment:
Why not skip _before here, you have them already in the folder "beforeCuts"?

Reply:
I wanted to keep it in case I quickly plot its still directly visible what it is. But if wanted I can of course remove the _before, let me know

Reply:
I meant to do: `beforeCuts/hPtRec", "hPtRec_before`
In this case it should be in the title, but not in the object name (but that one is always in the folder, so it is clear)